src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs

```diff
@@ -256,6 +256,16 @@ void Model_PropertyChanged(object sender, System.ComponentModel.PropertyChangedE
                     RaisePropertyChanged("RunEnabled");
                     RaisePropertyChanged("RunButtonEnabled");
                     RaisePropertyChanged("RunButtonToolTip");
+                    if (Application.Current != null)
+                    {
+                        Application.Current.Dispatcher.Invoke(new Action(() =>
+                        {
+                            dynamoViewModel.ShowOpenDialogAndOpenResultCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.NewHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.OpenRecentCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.CloseHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                        }));
+                    }
```

Comment:
You are absolutely correct - `RaisePropertyChanged` could be called in non-UI thread, but `RaiseCanExecuteChanged` has to be called in UI thread. The reason is, `RaiseCanExecuteChanged` might trigger some UI changes, for example, the enable/disable state of menu item or button, so it has to be called in UI thread (see [this StackOverflow answer](http://stackoverflow.com/questions/15075553/how-can-i-update-a-commands-canexecute-value-from-a-different-thread)).